// @assistant-ui/react/dist/runtimes/external-store/ExternalStoreRuntimeCore.mjs
// src/runtimes/external-store/ExternalStoreRuntimeCore.tsx
import { BaseAssistantRuntimeCore } from "../../internal.mjs";
import { ExternalStoreThreadListRuntimeCore } from "./ExternalStoreThreadListRuntimeCore.mjs";
import { ExternalStoreThreadRuntimeCore } from "./ExternalStoreThreadRuntimeCore.mjs";
var getThreadListAdapter = (store) => {
  return {
    threadId: store.threadId,
    onSwitchToNewThread: store.onSwitchToNewThread,
    onSwitchToThread: store.onSwitchToThread,
    ...store.adapters?.threadList
  };
};
var ExternalStoreRuntimeCore = class extends BaseAssistantRuntimeCore {
  threadList;
  _store;
  constructor(store) {
    super();
    this._store = store;
    this.threadList = new ExternalStoreThreadListRuntimeCore(
      getThreadListAdapter(store),
      (threadId) => new ExternalStoreThreadRuntimeCore(
        this._proxyConfigProvider,
        threadId,
        this._store
      )
    );
  }
  setStore(store) {
    this._store = store;
    this.threadList.setAdapter(getThreadListAdapter(store));
    this.threadList.mainThread.setStore(store);
  }
};
export {
  ExternalStoreRuntimeCore
};
