// @assistant-ui/react/dist/primitive-hooks/actionBar/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/primitive-hooks/actionBar/index.ts
var actionBar_exports = {};
__export(actionBar_exports, {
  useActionBarCopy: () => import_useActionBarCopy.useActionBarCopy,
  useActionBarEdit: () => import_useActionBarEdit.useActionBarEdit,
  useActionBarFeedbackNegative: () => import_useActionBarFeedbackNegative.useActionBarFeedbackNegative,
  useActionBarFeedbackPositive: () => import_useActionBarFeedbackPositive.useActionBarFeedbackPositive,
  useActionBarReload: () => import_useActionBarReload.useActionBarReload,
  useActionBarSpeak: () => import_useActionBarSpeak.useActionBarSpeak,
  useActionBarStopSpeaking: () => import_useActionBarStopSpeaking.useActionBarStopSpeaking
});
module.exports = __toCommonJS(actionBar_exports);
var import_useActionBarCopy = require("./useActionBarCopy.cjs");
var import_useActionBarEdit = require("./useActionBarEdit.cjs");
var import_useActionBarReload = require("./useActionBarReload.cjs");
var import_useActionBarSpeak = require("./useActionBarSpeak.cjs");
var import_useActionBarStopSpeaking = require("./useActionBarStopSpeaking.cjs");
var import_useActionBarFeedbackPositive = require("./useActionBarFeedbackPositive.cjs");
var import_useActionBarFeedbackNegative = require("./useActionBarFeedbackNegative.cjs");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  useActionBarCopy,
  useActionBarEdit,
  useActionBarFeedbackNegative,
  useActionBarFeedbackPositive,
  useActionBarReload,
  useActionBarSpeak,
  useActionBarStopSpeaking
});
