// @assistant-ui/react/dist/ui/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/ui/index.ts
var ui_exports = {};
__export(ui_exports, {
  AssistantActionBar: () => import_assistant_action_bar.default,
  AssistantMessage: () => import_assistant_message.default,
  AssistantModal: () => import_assistant_modal.default,
  AttachmentUI: () => import_attachment.default,
  BranchPicker: () => import_branch_picker.default,
  Composer: () => import_composer.default,
  ComposerAttachment: () => import_attachment.default,
  ContentPart: () => import_content_part.default,
  EditComposer: () => import_edit_composer.default,
  Thread: () => import_thread.default,
  ThreadConfigProvider: () => import_thread_config.ThreadConfigProvider,
  ThreadWelcome: () => import_thread_welcome.default,
  UserActionBar: () => import_user_action_bar.default,
  UserMessage: () => import_user_message.default,
  UserMessageAttachment: () => import_attachment.default,
  useThreadConfig: () => import_thread_config.useThreadConfig
});
module.exports = __toCommonJS(ui_exports);
var import_thread_config = require("./thread-config.cjs");
var import_assistant_action_bar = __toESM(require("./assistant-action-bar.cjs"));
var import_assistant_message = __toESM(require("./assistant-message.cjs"));
var import_assistant_modal = __toESM(require("./assistant-modal.cjs"));
var import_branch_picker = __toESM(require("./branch-picker.cjs"));
var import_composer = __toESM(require("./composer.cjs"));
var import_attachment = __toESM(require("./attachment.cjs"));
var import_edit_composer = __toESM(require("./edit-composer.cjs"));
var import_thread = __toESM(require("./thread.cjs"));
var import_user_message = __toESM(require("./user-message.cjs"));
var import_user_action_bar = __toESM(require("./user-action-bar.cjs"));
var import_thread_welcome = __toESM(require("./thread-welcome.cjs"));
var import_content_part = __toESM(require("./content-part.cjs"));
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AssistantActionBar,
  AssistantMessage,
  AssistantModal,
  AttachmentUI,
  BranchPicker,
  Composer,
  ComposerAttachment,
  ContentPart,
  EditComposer,
  Thread,
  ThreadConfigProvider,
  ThreadWelcome,
  UserActionBar,
  UserMessage,
  UserMessageAttachment,
  useThreadConfig
});
