// @assistant-ui/react/dist/primitives/message/MessageContent.js
"use strict";
"use client";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/primitives/message/MessageContent.tsx
var MessageContent_exports = {};
__export(MessageContent_exports, {
  MessagePrimitiveContent: () => MessagePrimitiveContent
});
module.exports = __toCommonJS(MessageContent_exports);
var import_react = require("react");
var import_context = require("../../context/index.cjs");
var import_MessageContext = require("../../context/react/MessageContext.cjs");
var import_ContentPartRuntimeProvider = require("../../context/providers/ContentPartRuntimeProvider.cjs");
var import_ContentPartText = require("../contentPart/ContentPartText.cjs");
var import_ContentPartImage = require("../contentPart/ContentPartImage.cjs");
var import_ContentPartDisplay = require("../contentPart/ContentPartDisplay.cjs");
var import_ContentPartInProgress = require("../contentPart/ContentPartInProgress.cjs");
var import_MessageRuntime = require("../../api/MessageRuntime.cjs");
var import_jsx_runtime = require("react/jsx-runtime");
var ToolUIDisplay = ({
  UI,
  ...props
}) => {
  const Render = (0, import_context.useToolUIs)((s) => s.getToolUI(props.toolName)) ?? UI;
  if (!Render) return null;
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(Render, { ...props });
};
var defaultComponents = {
  Text: () => /* @__PURE__ */ (0, import_jsx_runtime.jsxs)("p", { style: { whiteSpace: "pre-line" }, children: [
    /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_ContentPartText.ContentPartPrimitiveText, {}),
    /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_ContentPartInProgress.ContentPartPrimitiveInProgress, { children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)("span", { style: { fontFamily: "revert" }, children: " \u25CF" }) })
  ] }),
  Image: () => /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_ContentPartImage.ContentPartPrimitiveImage, {}),
  Unstable_Audio: () => null,
  UI: () => /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_ContentPartDisplay.ContentPartPrimitiveDisplay, {})
};
var MessageContentPartComponent = ({
  components: {
    Text = defaultComponents.Text,
    Empty,
    Image = defaultComponents.Image,
    UI = defaultComponents.UI,
    Unstable_Audio: Audio = defaultComponents.Unstable_Audio,
    tools: { by_name = {}, Fallback = void 0 } = {}
  } = {}
}) => {
  const contentPartRuntime = (0, import_context.useContentPartRuntime)();
  const part = (0, import_context.useContentPart)();
  const type = part.type;
  if (type === "tool-call") {
    const Tool = by_name[part.toolName] || Fallback;
    const addResult = (result) => contentPartRuntime.addToolResult(result);
    return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(ToolUIDisplay, { ...part, part, UI: Tool, addResult });
  }
  if (part.status.type === "requires-action")
    throw new Error("Encountered unexpected requires-action status");
  switch (type) {
    case "text":
      if (part.part === import_MessageRuntime.EMPTY_CONTENT && !!Empty) {
        return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(Empty, { status: part.status });
      }
      return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(Text, { ...part, part });
    case "image":
      return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(Image, { ...part, part });
    case "audio":
      return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(Audio, { ...part, part });
    case "ui":
      return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(UI, { ...part, part });
    default:
      const unhandledType = type;
      throw new Error(`Unknown content part type: ${unhandledType}`);
  }
};
var MessageContentPartImpl = ({
  partIndex,
  components
}) => {
  const messageRuntime = (0, import_MessageContext.useMessageRuntime)();
  const runtime = (0, import_react.useMemo)(
    () => messageRuntime.getContentPartByIndex(partIndex),
    [messageRuntime, partIndex]
  );
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_ContentPartRuntimeProvider.ContentPartRuntimeProvider, { runtime, children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(MessageContentPartComponent, { components }) });
};
var MessageContentPart = (0, import_react.memo)(
  MessageContentPartImpl,
  (prev, next) => prev.partIndex === next.partIndex && prev.components?.Text === next.components?.Text && prev.components?.Image === next.components?.Image && prev.components?.Unstable_Audio === next.components?.Unstable_Audio && prev.components?.UI === next.components?.UI && prev.components?.tools === next.components?.tools
);
var MessagePrimitiveContent = ({
  components
}) => {
  const contentLength = (0, import_MessageContext.useMessage)((s) => s.content.length) || 1;
  return Array.from({ length: contentLength }, (_, index) => /* @__PURE__ */ (0, import_jsx_runtime.jsx)(MessageContentPart, { partIndex: index, components }, index));
};
MessagePrimitiveContent.displayName = "MessagePrimitive.Content";
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  MessagePrimitiveContent
});
