// @assistant-ui/react/dist/internal.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __reExport = (target, mod, secondTarget) => (__copyProps(target, mod, "default"), secondTarget && __copyProps(secondTarget, mod, "default"));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/internal.ts
var internal_exports = {};
__export(internal_exports, {
  AssistantRuntimeImpl: () => import_AssistantRuntime.AssistantRuntimeImpl,
  BaseAssistantRuntimeCore: () => import_BaseAssistantRuntimeCore.BaseAssistantRuntimeCore,
  DefaultThreadComposerRuntimeCore: () => import_DefaultThreadComposerRuntimeCore.DefaultThreadComposerRuntimeCore,
  MessageRepository: () => import_MessageRepository.MessageRepository,
  ProxyConfigProvider: () => import_ProxyConfigProvider.ProxyConfigProvider,
  ThreadRuntimeImpl: () => import_ThreadRuntime.ThreadRuntimeImpl,
  TooltipIconButton: () => import_tooltip_icon_button.TooltipIconButton,
  generateId: () => import_idUtils.generateId
});
module.exports = __toCommonJS(internal_exports);
var import_DefaultThreadComposerRuntimeCore = require("./runtimes/composer/DefaultThreadComposerRuntimeCore.cjs");
var import_ProxyConfigProvider = require("./utils/ProxyConfigProvider.cjs");
var import_MessageRepository = require("./runtimes/utils/MessageRepository.cjs");
var import_BaseAssistantRuntimeCore = require("./runtimes/core/BaseAssistantRuntimeCore.cjs");
var import_tooltip_icon_button = require("./ui/base/tooltip-icon-button.cjs");
var import_idUtils = require("./utils/idUtils.cjs");
var import_AssistantRuntime = require("./api/AssistantRuntime.cjs");
var import_ThreadRuntime = require("./api/ThreadRuntime.cjs");
__reExport(internal_exports, require("./utils/smooth/index.cjs"), module.exports);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AssistantRuntimeImpl,
  BaseAssistantRuntimeCore,
  DefaultThreadComposerRuntimeCore,
  MessageRepository,
  ProxyConfigProvider,
  ThreadRuntimeImpl,
  TooltipIconButton,
  generateId,
  ...require("./utils/smooth/index.cjs")
});
