// @assistant-ui/react/dist/context/react/AssistantContext.mjs
"use client";

// src/context/react/AssistantContext.ts
import { createContext } from "react";
import { createContextHook } from "./utils/createContextHook.mjs";
import { createContextStoreHook } from "./utils/createContextStoreHook.mjs";
var AssistantContext = createContext(
  null
);
var useAssistantContext = createContextHook(
  AssistantContext,
  "AssistantRuntimeProvider"
);
function useAssistantRuntime(options) {
  const context = useAssistantContext(options);
  if (!context) return null;
  return context.useAssistantRuntime();
}
var actions = createContextStoreHook(
  useAssistantContext,
  "useAssistantActions"
);
var useAssistantActionsStore = actions.useAssistantActionsStore;
var useAssistantActions = actions.useAssistantActions;
var useAssistantRuntimeStore = useAssistantActionsStore;
var { useToolUIs, useToolUIsStore } = createContextStoreHook(
  useAssistantContext,
  "useToolUIs"
);
var { useThreadList } = createContextStoreHook(
  useAssistantContext,
  "useThreadList"
);
export {
  AssistantContext,
  actions,
  useAssistantActions,
  useAssistantActionsStore,
  useAssistantContext,
  useAssistantRuntime,
  useAssistantRuntimeStore,
  useThreadList,
  useToolUIs,
  useToolUIsStore
};
