// @assistant-ui/react/dist/context/react/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/context/react/index.ts
var react_exports = {};
__export(react_exports, {
  useAssistantActions: () => import_AssistantContext.useAssistantActions,
  useAssistantActionsStore: () => import_AssistantContext.useAssistantActionsStore,
  useAssistantContext: () => import_AssistantContext.useAssistantContext,
  useAssistantRuntime: () => import_AssistantContext.useAssistantRuntime,
  useAssistantRuntimeStore: () => import_AssistantContext.useAssistantRuntimeStore,
  useAttachment: () => import_AttachmentContext.useAttachment,
  useAttachmentRuntime: () => import_AttachmentContext.useAttachmentRuntime,
  useComposer: () => import_ComposerContext.useComposer,
  useComposerContext: () => import_ComposerContext.useComposerContext,
  useComposerRuntime: () => import_ComposerContext.useComposerRuntime,
  useComposerStore: () => import_ComposerContext.useComposerStore,
  useContentPart: () => import_ContentPartContext.useContentPart,
  useContentPartContext: () => import_ContentPartContext.useContentPartContext,
  useContentPartRuntime: () => import_ContentPartContext.useContentPartRuntime,
  useContentPartStore: () => import_ContentPartContext.useContentPartStore,
  useEditComposer: () => import_MessageContext.useEditComposer,
  useEditComposerStore: () => import_MessageContext.useEditComposerStore,
  useMessage: () => import_MessageContext.useMessage,
  useMessageContext: () => import_MessageContext.useMessageContext,
  useMessageRuntime: () => import_MessageContext.useMessageRuntime,
  useMessageStore: () => import_MessageContext.useMessageStore,
  useMessageUtils: () => import_MessageContext.useMessageUtils,
  useMessageUtilsStore: () => import_MessageContext.useMessageUtilsStore,
  useThread: () => import_ThreadContext.useThread,
  useThreadActions: () => import_ThreadContext.useThreadActions,
  useThreadActionsStore: () => import_ThreadContext.useThreadActionsStore,
  useThreadComposer: () => import_ThreadContext.useThreadComposer,
  useThreadComposerStore: () => import_ThreadContext.useThreadComposerStore,
  useThreadContext: () => import_ThreadContext.useThreadContext,
  useThreadList: () => import_AssistantContext.useThreadList,
  useThreadMessages: () => import_ThreadContext.useThreadMessages,
  useThreadMessagesStore: () => import_ThreadContext.useThreadMessagesStore,
  useThreadModelConfig: () => import_ThreadContext.useThreadModelConfig,
  useThreadRuntime: () => import_ThreadContext.useThreadRuntime,
  useThreadRuntimeStore: () => import_ThreadContext.useThreadRuntimeStore,
  useThreadStore: () => import_ThreadContext.useThreadStore,
  useThreadViewport: () => import_ThreadContext.useThreadViewport,
  useThreadViewportStore: () => import_ThreadContext.useThreadViewportStore,
  useToolUIs: () => import_AssistantContext.useToolUIs,
  useToolUIsStore: () => import_AssistantContext.useToolUIsStore
});
module.exports = __toCommonJS(react_exports);
var import_AssistantContext = require("./AssistantContext.cjs");
var import_ThreadContext = require("./ThreadContext.cjs");
var import_MessageContext = require("./MessageContext.cjs");
var import_ContentPartContext = require("./ContentPartContext.cjs");
var import_ComposerContext = require("./ComposerContext.cjs");
var import_AttachmentContext = require("./AttachmentContext.cjs");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  useAssistantActions,
  useAssistantActionsStore,
  useAssistantContext,
  useAssistantRuntime,
  useAssistantRuntimeStore,
  useAttachment,
  useAttachmentRuntime,
  useComposer,
  useComposerContext,
  useComposerRuntime,
  useComposerStore,
  useContentPart,
  useContentPartContext,
  useContentPartRuntime,
  useContentPartStore,
  useEditComposer,
  useEditComposerStore,
  useMessage,
  useMessageContext,
  useMessageRuntime,
  useMessageStore,
  useMessageUtils,
  useMessageUtilsStore,
  useThread,
  useThreadActions,
  useThreadActionsStore,
  useThreadComposer,
  useThreadComposerStore,
  useThreadContext,
  useThreadList,
  useThreadMessages,
  useThreadMessagesStore,
  useThreadModelConfig,
  useThreadRuntime,
  useThreadRuntimeStore,
  useThreadStore,
  useThreadViewport,
  useThreadViewportStore,
  useToolUIs,
  useToolUIsStore
});
