// @assistant-ui/react/dist/context/react/MessageContext.js
"use strict";
"use client";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/context/react/MessageContext.ts
var MessageContext_exports = {};
__export(MessageContext_exports, {
  MessageContext: () => MessageContext,
  useEditComposer: () => useEditComposer,
  useEditComposerStore: () => useEditComposerStore,
  useMessage: () => useMessage,
  useMessageContext: () => useMessageContext,
  useMessageRuntime: () => useMessageRuntime,
  useMessageStore: () => useMessageStore,
  useMessageUtils: () => useMessageUtils,
  useMessageUtilsStore: () => useMessageUtilsStore
});
module.exports = __toCommonJS(MessageContext_exports);
var import_react = require("react");
var import_createContextHook = require("./utils/createContextHook.cjs");
var import_createContextStoreHook = require("./utils/createContextStoreHook.cjs");
var MessageContext = (0, import_react.createContext)(null);
var useMessageContext = (0, import_createContextHook.createContextHook)(
  MessageContext,
  "a component passed to <ThreadPrimitive.Messages components={...} />"
);
function useMessageRuntime(options) {
  const context = useMessageContext(options);
  if (!context) return null;
  return context.useMessageRuntime();
}
var { useMessage, useMessageStore } = (0, import_createContextStoreHook.createContextStoreHook)(
  useMessageContext,
  "useMessage"
);
var { useMessageUtils, useMessageUtilsStore } = (0, import_createContextStoreHook.createContextStoreHook)(
  useMessageContext,
  "useMessageUtils"
);
var { useEditComposer, useEditComposerStore } = (0, import_createContextStoreHook.createContextStoreHook)(
  useMessageContext,
  "useEditComposer"
);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  MessageContext,
  useEditComposer,
  useEditComposerStore,
  useMessage,
  useMessageContext,
  useMessageRuntime,
  useMessageStore,
  useMessageUtils,
  useMessageUtilsStore
});
