// @assistant-ui/react/dist/context/react/MessageContext.mjs
"use client";

// src/context/react/MessageContext.ts
import { createContext } from "react";
import { createContextHook } from "./utils/createContextHook.mjs";
import { createContextStoreHook } from "./utils/createContextStoreHook.mjs";
var MessageContext = createContext(null);
var useMessageContext = createContextHook(
  MessageContext,
  "a component passed to <ThreadPrimitive.Messages components={...} />"
);
function useMessageRuntime(options) {
  const context = useMessageContext(options);
  if (!context) return null;
  return context.useMessageRuntime();
}
var { useMessage, useMessageStore } = createContextStoreHook(
  useMessageContext,
  "useMessage"
);
var { useMessageUtils, useMessageUtilsStore } = createContextStoreHook(
  useMessageContext,
  "useMessageUtils"
);
var { useEditComposer, useEditComposerStore } = createContextStoreHook(
  useMessageContext,
  "useEditComposer"
);
export {
  MessageContext,
  useEditComposer,
  useEditComposerStore,
  useMessage,
  useMessageContext,
  useMessageRuntime,
  useMessageStore,
  useMessageUtils,
  useMessageUtilsStore
};
